core/shared/src/main/scala/result.scala

```diff
@@ -170,6 +170,17 @@ sealed abstract class Result[+T, E <: Exception](val answer: T, val errors: Seq[
       case Unforeseen(e) => throw e
     }
 
+  /** Filter on the answer of this result. */
+  def filter(p: T => Boolean): Result[T, E] =
+    this match {
+      case Errata(_) => this
+      case Answer(b) => if(p(b)) this else Errata[T, E](Nil)
```

Comment:
Not sure we should be producing an `Errata` or `Nil`. I may have elsewhere made some assumptions that it's non-empty. There's no justification for it being `Unforeseen`, so could it be an `Errata(SomeException)` (and we track the exception in the type, i.e.

```
def filter(p: T => Boolean): Result[T, E with SomeException]
```

Any idea what that exception should be called? I think it's reasonable to invent a new one, specifically for this purpose.